server/src/main/java/org/opensearch/identity/tokens/TokenManager.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.tokens;
+
+/**
+ * This interface defines the expected methods of a token manager
+ */
+public interface TokenManager {
```

Comment:
If an identity plugin needs to implement the methods of this interface then similar to `getSubject()` would we need to add a `getAuthTokenManager` or `getTokenManager` to the IdentityPlugin interface?

Since the work token is pretty prevalent in OpenSearch, do you think the class name should be more specific like `AuthTokenManager`?